src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,289 @@
+                      std::is_same<TReg, Xbyak::Reg64>::value || std::is_same<TReg, Xbyak::Reg32>::value || std::is_same<TReg, Xbyak::Opmask>::value,
+                      "Unsupported TReg by RegistersPool. Please, use the following registers either Reg32, Reg64, Xmm, Ymm, Zmm or Opmask");
+        if (std::is_same<TReg, Xbyak::Xmm>::value || std::is_same<TReg, Xbyak::Ymm>::value || std::is_same<TReg, Xbyak::Zmm>::value) {
+            auto idx = simdSet.getUnused(fixedIdx);
+            simdSet.setAsUsed(idx);
+            return idx;
+        } else if (std::is_same<TReg, Xbyak::Reg64>::value || std::is_same<TReg, Xbyak::Reg32>::value) {
+            auto idx = generalSet.getUnused(fixedIdx);
+            generalSet.setAsUsed(idx);
+            return idx;
+        } else if (std::is_same<TReg, Xbyak::Opmask>::value) {
+            return getFreeOpmask(fixedIdx);
+        }
+    }
+
+    template<typename TReg>
+    void returnToPool(TReg reg) {
+        if (std::is_same<TReg, Xbyak::Xmm>::value || std::is_same<TReg, Xbyak::Ymm>::value || std::is_same<TReg, Xbyak::Zmm>::value) {
+            simdSet.setAsUnused(reg.getIdx());
+        } else if (std::is_same<TReg, Xbyak::Reg64>::value || std::is_same<TReg, Xbyak::Reg32>::value) {
+            generalSet.setAsUnused(reg.getIdx());
+        } else if (std::is_same<TReg, Xbyak::Opmask>::value) {
+            returnOpmaskToPool(reg.getIdx());
+        }
+    }
+
+    void checkUniqueAndUpdate(bool isCtor = true) {
+        static thread_local bool isCreated = false;
+        if (isCtor) {
+            if (isCreated) {
+                IE_THROW() << "There should be only one instance of RegistersPool per thread";
+            }
+            isCreated = true;
+        } else {
+            isCreated = false;
+        }
+    }
+
+    PhysicalSet generalSet {16};
+    PhysicalSet simdSet;
+};
+
+template <x64::cpu_isa_t isa>
+class IsaRegistersPool : public RegistersPool {
+public:
+    IsaRegistersPool(std::initializer_list<Xbyak::Reg> regsToExclude) : RegistersPool(regsToExclude, x64::cpu_isa_traits<isa>::n_vregs) {}
+};
+
+template <>
+class IsaRegistersPool<x64::avx512_core> : public RegistersPool {
+public:
+    IsaRegistersPool(std::initializer_list<Xbyak::Reg> regsToExclude)
+            : RegistersPool(regsToExclude, x64::cpu_isa_traits<x64::avx512_core>::n_vregs) {}
+
+    int getFreeOpmask(int fixedIdx) override {
+        auto idx = opmaskSet.getUnused(fixedIdx);
+        opmaskSet.setAsUsed(idx);
+        return idx;
+    }
+
+    void returnOpmaskToPool(int idx) override {
+        opmaskSet.setAsUnused(idx);
+    }
+
+    void excludeOpmask(const Xbyak::Opmask& reg) override {
+        opmaskSet.exclude(reg);
+    }
+
+    size_t countUnusedOpmask() const override {
+        return opmaskSet.countUnused();
+    }
+
+protected:
+    PhysicalSet opmaskSet {8};
+};
+
+template <x64::cpu_isa_t isa>
+RegistersPool::Ptr RegistersPool::create(std::initializer_list<Xbyak::Reg> regsToExclude) {
```

Comment:
1. why the pool with the argument "regsToExclude" is not covered in unit test?
2. a general question to the test - can you use type-parameterized test suite to cover different ISAs?

Reply:
Thanks, I added the unit test for `regsToExclude`. Yes, testing for different `isa` is a good idea and it could be done after this solution is accepted.

Reply:
> Thanks, I added the unit test for `regsToExclude`. Yes, testing for different `isa` is a good idea and it could be done after this solution is accepted.

I think we can add unit tests for different isa now.
For example using `mayiuse()` functionality

Reply:
@ceciliapeng2011 I added the type-parameterized test suite to cover different ISA and different Xbyak register types.